src/app/Library/CrudPanel/Uploads/Uploaders/Uploader.php

```diff
@@ -0,0 +1,386 @@
+<?php
+
+namespace Backpack\CRUD\app\Library\CrudPanel\Uploads\Uploaders;
+
+use Backpack\CRUD\app\Library\CrudPanel\CrudPanelFacade as CRUD;
+use Backpack\CRUD\app\Library\CrudPanel\Uploads\Interfaces\UploaderInterface;
+use Backpack\CRUD\app\Library\CrudPanel\Uploads\Traits\HasCrudObjectType;
+use Backpack\CRUD\app\Library\CrudPanel\Uploads\Traits\HasName;
+use Closure;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Support\Facades\Storage;
+use Illuminate\Support\Str;
+
+abstract class Uploader implements UploaderInterface
+{
+    use HasCrudObjectType, HasName;
+
+    /**
+     * Indicates the uploaded file should be deleted when entry is deleted.
+     *
+     * @var bool
+     */
+    public $deleteWhenEntryIsDeleted = true;
+
+    /**
+     * Indicates if this uploader instance is inside a repeatable container.
+     *
+     * @var bool
+     */
+    public $isRepeatable = false;
+
+    /**
+     * When inside a repeatable container, indicates the container name.
+     *
+     * @var string|null
+     */
+    public $repeatableContainerName = null;
+
+    /**
+     * Developer provided filename.
+     *
+     * @var null|string|Closure
+     */
+    public $fileName = null;
+
+    /**
+     * The disk where upload will be stored. By default `public`.
+     *
+     * @var string
+     */
+    public $disk = 'public';
+
+    /**
+     * Indicates if the upload handles multiple files.
+     *
+     * @var bool
+     */
+    public $isMultiple = false;
+
+    /**
+     * The class of the entry where uploads will be attached to.
+     *
+     * @var string
+     */
+    public $entryClass;
+
+    /**
+     * The path inside the disk to store the uploads.
+     *
+     * @var string
+     */
+    public $path = '';
+
+    /**
+     * Should the url to the object be a temporary one (eg: s3).
+     *
+     * @var bool
+     */
+    public $temporary = false;
+
+    /**
+     * When using temporary urls, defines the time that the url
+     * should be available in minutes.
+     *
+     * By default 1 minute
+     *
+     * @var int
+     */
+    public $expiration = 1;
```

Comment:
If it's only used for temporary URLs, it should maybe say that in the name:

```suggestion
    public $temporaryUrlExpirationTime = 1;
```